### PR TITLE
Add func to XCTestCase for waiting for dispatch groups

### PR DESCRIPTION
**Pull request by typfel**
## What's new in this PR?

Add helper method for waiting for dispatch groups in tests.

### CODE REVIEW

File: Source/Public/XCTestCase+Helpers.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+//
+
+import Foundation
+
+extension XCTestCase {
+    
+    public func waitForGroupsToBeEmpty(_ groups: [DispatchGroup], timeout: TimeInterval = 5) -> Bool {
+        
+        let timeoutDate = Date(timeIntervalSinceNow: timeout)
+        var groupCounter = groups.count
```
Comment (mikeger) [inline Source/Public/XCTestCase+Helpers.swift]: I think you can create another group to keep track of multiple other groups 💥

Comment (typfel) [reply Source/Public/XCTestCase+Helpers.swift]: Which API are you referring to? Remember that we don't want to block the main queue while we are waiting.

Comment (typfel) [reply Source/Public/XCTestCase+Helpers.swift]: Yes that would probably work.

Comment (typfel) [reply Source/Public/XCTestCase+Helpers.swift]: But is it easier to understand? I basically ported a similar function from `ZMBaseTest`

Comment (mikeger) [reply Source/Public/XCTestCase+Helpers.swift]: I don't think it matter as long as all the groups are notifying on the main queue (so original implementation is good).

Comment (typfel) [reply Source/Public/XCTestCase+Helpers.swift]: 👍 So you'll add that ✅ ? :)

**Comment by typfel** [comment]
@zenkins release minor

**Pull request merged**